src/flake8_tidy_imports/settings.rs

```diff
@@ -31,17 +41,35 @@ pub struct Options {
     /// Whether to ban all relative imports (`"all"`), or only those imports
     /// that extend into the parent module and beyond (`"parents"`).
     pub ban_relative_imports: Option<Strictness>,
+
+    #[option(
+        default = r#"{}"#,
+        value_type = "HashMap<String, BannedApi>",
+        example = r#"
+            [tool.ruff.flake8-tidy-imports.banned-api]
+            "cgi".msg = "The cgi module is deprecated, see https://peps.python.org/pep-0594/#cgi."
+            "typing.TypedDict".msg = "Use typing_extensions.TypedDict instead."
+        "#
+    )]
+    /// Specific modules or module members that may not be imported/accessed.
+    ///
+    /// Note that this check is only meant to flag accidental uses,
+    /// it can be easily circumvented via `eval` or `importlib` and
+    /// attempting to ban those via this setting is a futile endeavor.
+    pub banned_api: FxHashMap<String, BannedApi>,
 }
 
-#[derive(Debug, Hash)]
+#[derive(Debug)]
 pub struct Settings {
     pub ban_relative_imports: Strictness,
+    pub banned_api: FxHashMap<String, BannedApi>,
```

Comment:
I know you explicitly preferred `banned_api`, but given the minor difference, it's worth it to me to maintain compatibility with `flake8-tidy-imports` and keep this as `banned-modules`. How do you feel about that?

Reply:
Yes I don't like `banned-modules` because it's misleading since the setting can also be used for module members. I do think correct semantics make a big difference in terms of usability. And I don't get your point about compatibility given that you cannot copy'n'paste `flake8-tidy-imports` config into the ruff config anyway because:

1. flake8-tidy-imports does not use TOML but .cfg which uses a different syntax.
2. You have to wrap the names in double quotes and appent `.msg`.

If we still want both settings to have the same name despite having different formats, I think it would make more sense if `flake8-tidy-imports` updated its setting name to `banned-api` since it's the more correct terminology.

Reply:
I guess this is fine. There is some mental overhead to deviating (folks that migrate their config have to know to change the setting name) but I understand your point, and I don't feel strongly.

What's your reaction to `banned import`? There was some discussion [here](https://github.com/adamchainz/flake8-tidy-imports/issues/38) and it looks like that's the verbiage that `flake8-tidy-imports` uses in their [messages](https://github.com/adamchainz/flake8-tidy-imports/pull/40).

Reply:
I was thinking about this a bit more, and I prefer `banned-imports` over `banned-api`. The issue with `banned-api` is that it's _too_ general, and actually implies things that go beyond what the tool can do. For example, to me, `banned-api` would also include things like method calls.

Reply:
(If you agree, I'm also happy to do the work of renaming, then merge the PR.)

Reply:
Yes "banned import" would be accurate for `flake8-tidy-imports` but only because they don't yet implement the attribute access check (adamchainz/flake8-tidy-imports/issues/63), which this PR implements. Since the check also checks for attribute access "banned import" isn't accurate, for example:

```python
import foo
foo.bar # TIDI201 Banned import 'foo.bar' used
```

`foo.bar` here is not an import.